plugin/pkg/shoot/validator/admission.go

```diff
@@ -515,6 +515,14 @@ func (c *validationContext) validateShootNetworks() field.ErrorList {
 			}
 		}
 
+		// validate network disjointedness within shoot network
+		allErrs = append(allErrs, cidrvalidation.ValidateShootNetworkDisjointedness(
+			path,
+			c.shoot.Spec.Networking.Nodes,
+			c.shoot.Spec.Networking.Pods,
+			c.shoot.Spec.Networking.Services,
+		)...)
```

Comment:
This function can probably called in the "static validation" in `pkg/apis/core/validation/shoot_test.go`. Or is there a particular reason why you put it into the "dynamic validation"?

Reply:
Thanks, I didn't check it before that `Networking` field is immutable.

Reply:
I think we should keep it here only because in `pkg/apis/core/validation/shoot.go`, `Spec.Networking.Pods` and `Spec.Networking.Services` can be `nil`. https://github.com/gardener/gardener/blob/6159c839ded65dc64ab9bc8f742ee9117efc7642/pkg/apis/core/validation/shoot.go#L806-L838

But here we are upadating these fields based on `ShootDefaults` also. https://github.com/gardener/gardener/blob/6159c839ded65dc64ab9bc8f742ee9117efc7642/plugin/pkg/shoot/validator/admission.go#L502-L516.

Reply:
Right, that makes perfect sense. Let's leave it there! Thank you :)